eth_p2p/rlpx.nim

```diff
@@ -1108,7 +1108,7 @@ rlpxProtocol p2p(version = 0):
     discard
 
 proc removePeer(network: EthereumNode, peer: Peer) =
-  if network.peerPool != nil:
+  if network.peerPool != nil and not peer.remote.isNil:
```

Comment:
Can you clarify with a comment in the code what conditions triggers the `peer.remote.isNil` check?